src/components/common/Extensions/MainNavigation/MainNavigation.tsx

```diff
@@ -1,42 +1,103 @@
+import {
+  useAppContext,
+  useColonyContext,
+  useMobile,
+  useUserReputation,
+} from '~hooks';
 import { SubNavigationMobile } from '~common/Extensions/SubNavigation';
 import { LEARN_MORE_PAYMENTS } from '~constants';
 import LearnMore from '~shared/Extensions/LearnMore';
 import Button from '~shared/Extensions/Button';
+import { MainNavigationProps } from './types';
+import PopoverBase from '~shared/Extensions/PopoverBase';
+import UserAvatar from '~shared/Extensions/UserAvatar';
+import MemberReputation from '../UserNavigation/partials/MemberReputation';
+import Icon from '~shared/Icon';
+import Token from '../UserNavigation/partials/Token';
+import styles from './MainNavigation.module.css';
 
 const displayName = 'common.Extensions.MainNavigation';
 
-const MainNavigation: FC = () => {
+const MainNavigation: FC<MainNavigationProps> = ({
+  setTooltipRef,
+  tooltipProps,
+  isMenuOpen,
+}) => {
+  const { colony } = useColonyContext();
+  const { formatMessage } = useIntl();
   const isMobile = useMobile();
+  const { user, wallet } = useAppContext();
+  const { profile } = user || {};
+  const { colonyAddress, nativeToken } = colony || {};
+  const { userReputation, totalReputation } = useUserReputation(
+    colonyAddress,
+    wallet?.address,
+  );
 
   return (
     <div className="py-6 sm:py-0">
-      <Nav items={navMenuItems} />
-      {isMobile && (
-        <>
-          <div className="border-t border-gray-200 mx-6 mb-3" />
-          <SubNavigationMobile />
-          <div className="flex flex-col items-center justify-between mx-6 border-t border-gray-200 mt-4">
-            <div className="mb-6 mt-6">
-              <Button
-                text="Create new action"
-                mode="secondaryOutline"
-                isFullSize={isMobile}
+      <div className="hidden sm:block">
+        <Nav items={navMenuItems} />
+      </div>
+      {isMobile && isMenuOpen && (
+        <PopoverBase
+          setTooltipRef={setTooltipRef}
+          tooltipProps={tooltipProps}
+          classNames="w-full border-none shadow-none px-0 pb-6"
+        >
+          <div className={styles.mobileButtons}>
+            {nativeToken && <Token nativeToken={nativeToken} />}
+            <Button mode="tertiaryOutline" isFullRounded>
+              <div className="flex items-center gap-3">
+                <UserAvatar
+                  userName={profile?.displayName || user?.name || ''}
+                  size="xxs"
+                  user={user}
+                />
+                <MemberReputation
+                  userReputation={userReputation}
+                  totalReputation={totalReputation}
+                  hideOnMobile={false}
+                />
+              </div>
+            </Button>
+            <Button mode="tertiaryOutline" isFullRounded>
+              <Icon name="list" appearance={{ size: 'extraTiny' }} />
+              {!isMobile && (
+                <p className="text-sm font-inter font-medium ml-1">
```

Comment:
Do we need these styles here`text-sm font-inter font-medium`?

Reply:
Whole <p> removed